packages/wrangler/src/api/pages/create-worker-bundle-contents.ts

```diff
@@ -43,43 +43,51 @@ function createWorkerBundleFormData(workerBundle: BundleResult): FormData {
 		type: workerBundle.bundleType || "esm",
 	};
 
+	const bindings: CfWorkerInit["bindings"] = {
+		kv_namespaces: config?.kv_namespaces,
+		vars: config?.vars,
+		browser: config?.browser,
+		ai: config?.ai,
+		durable_objects: config?.durable_objects,
+		queues: config?.queues.producers?.map((producer) => {
+			return { binding: producer.binding, queue_name: producer.queue };
+		}),
+		r2_buckets: config?.r2_buckets,
+		d1_databases: config?.d1_databases,
+		vectorize: config?.vectorize,
+		hyperdrive: config?.hyperdrive,
+		services: config?.services,
+		analytics_engine_datasets: config?.analytics_engine_datasets,
+		mtls_certificates: config?.mtls_certificates,
+		send_email: undefined,
+		wasm_modules: undefined,
+		text_blobs: undefined,
+		data_blobs: undefined,
+		constellation: undefined,
+		dispatch_namespaces: undefined,
+		logfwdr: undefined,
+		unsafe: undefined,
+	};
+
+	// The upload API only accepts an empty string or no specified placement for the "off" mode.
+	const placement: CfPlacement | undefined =
+		config?.placement?.mode === "smart" ? { mode: "smart" } : undefined;
+
 	const worker: CfWorkerInit = {
 		name: mainModule.name,
 		main: mainModule,
 		modules: workerBundle.modules,
-		bindings: {
-			vars: undefined,
-			kv_namespaces: undefined,
-			send_email: undefined,
-			wasm_modules: undefined,
-			text_blobs: undefined,
-			browser: undefined,
-			ai: undefined,
-			data_blobs: undefined,
-			durable_objects: undefined,
-			queues: undefined,
-			r2_buckets: undefined,
-			d1_databases: undefined,
-			vectorize: undefined,
-			constellation: undefined,
-			hyperdrive: undefined,
-			services: undefined,
-			analytics_engine_datasets: undefined,
-			dispatch_namespaces: undefined,
-			mtls_certificates: undefined,
-			logfwdr: undefined,
-			unsafe: undefined,
-		},
+		bindings,
 		migrations: undefined,
-		compatibility_date: undefined,
-		compatibility_flags: undefined,
+		compatibility_date: config?.compatibility_date,
+		compatibility_flags: config?.compatibility_flags,
 		usage_model: undefined,
 		keepVars: undefined,
 		keepSecrets: undefined,
-		logpush: undefined,
-		placement: undefined,
+		logpush: config?.logpush,
```

Comment:
should we set this to `undefined` since we're not supporting `lopush` for the time being?